python_modules/dagster/dagster/_core/definitions/asset_graph_differ.py

```diff
@@ -28,6 +27,79 @@ class AssetDefinitionChangeType(Enum):
     REMOVED = "REMOVED"
 
 
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailNew:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.NEW
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailCodeVersion:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.CODE_VERSION
+    base_value: Optional[str] = None
+    branch_value: Optional[str] = None
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailDependencies:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.DEPENDENCIES
+    added: Sequence[AssetKey] = []
+    changed: Sequence[AssetKey] = []
+    removed: Sequence[AssetKey] = []
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailPartitionsDefinition:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.PARTITIONS_DEFINITION
+    base_value: Optional[str] = None
```

Comment:
In what situation would it make sense for these to be None?

Reply:
The diff value for a partitions definition can be `None` if the asset does not have a partitions definition defined. However, I do think it's unnecessary to have a default value here as it should be explicitly set.